utils/python/CIME/BuildTools/configure.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+
+"""This script writes CIME build information to a directory.
+
+The pieces of information that will be written include:
+
+1. Machine-specific build settings (i.e. the "Macros" file).
+2. File-specific build settings (i.e. "Depends" files).
+3. Environment variable loads (i.e. the env_mach_specific files).
+
+The .env_mach_specific.sh and .env_mach_specific.csh files are specific to a
+given compiler, MPI library, and DEBUG setting. By default, these will be the
+machine's default compiler, the machine's default MPI library, and FALSE,
+respectively. These can be changed by setting the environment variables
+COMPILER, MPILIB, and DEBUG, respectively.
+"""
+
+import shutil
+from CIME.XML.standard_module_setup import *
+from CIME.utils import expect
+from CIME.XML.compilers import Compilers
+from CIME.XML.env_mach_specific import EnvMachSpecific
+
+logger = logging.getLogger(__name__)
+
+def configure(machobj, output_dir, macros_format, compiler, mpilib, debug, sysos):
+    """Add Macros, Depends, and env_mach_specific files to a directory.
+
+    Arguments:
+    machobj - Machines argument for this machine.
+    output_dir - Directory in which to place output.
+    macros_format - Container containing the string 'Makefile' to produce
+                    Makefile Macros output, and/or 'CMake' for CMake output.
+    compiler - String containing the compiler vendor to configure for.
+    mpilib - String containing the MPI implementation to configure for.
+    debug - Boolean specifying whether debugging options are enabled.
+    """
+    # Macros generation.
+    suffixes = {'Makefile': 'make', 'CMake': 'cmake'}
+    macro_maker = Compilers(machobj)
+    for form in macros_format:
```

Comment:
Why not just use macro_format directly rather than an for statement and simply check if macro_format is not None?